File: php/src/ValidationException.php
<?php

declare(strict_types=1);

namespace App;

use RuntimeException;

/**
 * Fachlicher Validierungsfehler -> wird im HTTP-Handler zu 400.
 * Trägt eine Liste der konkreten Feld-Fehler.
 */
final class ValidationException extends RuntimeException
{
    /** @param array<int,string> $errors */
    public function __construct(public readonly array $errors)
    {
        parent::__construct('Validierung fehlgeschlagen');
    }
}

File: java/src/main/java/de/train/bewerbung/ValidationException.java
package de.train.bewerbung;

import java.util.List;

/**
 * Fachlicher Validierungsfehler -> wird im HTTP-Handler zu 400.
 * Traegt eine Liste der konkreten Feld-Fehler.
 */
public class ValidationException extends RuntimeException {
    private final List<String> errors;

    public ValidationException(List<String> errors) {
        super("Validierung fehlgeschlagen");
        this.errors = errors;
    }

    public List<String> getErrors() {
        return errors;
    }
}
